File: webapp/src/main/vueapp/src/logic/models/Patient.ts
import {GenericEntity} from '~/logic/models/utils/GenericEntity';

export class Patient extends GenericEntity<Patient> {
    private _id: number;
    private _userId: number;
    private _firstName: string;
    private _surname: string;
    private _officeId: number;

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get userId(): number {
        return this._userId;
    }

    public set userId(value: number) {
        this._userId = value;
    }

    public get firstName(): string {
        return this._firstName;
    }

    public set firstName(value: string) {
        this._firstName = value;
    }

    public get surname(): string {
        return this._surname;
    }

    public set surname(value: string) {
        this._surname = value;
    }

    public get officeId(): number {
        return this._officeId;
    }

    public set officeId(value: number) {
        this._officeId = value;
    }
}

File: models/src/main/java/ar/edu/itba/paw/models/Patient.java
package ar.edu.itba.paw.models;

import javax.persistence.*;

@Entity
@Table(
        name = "patient",
        indexes = {
                @Index(columnList = "patient_id", name = "patient_patient_id_uindex", unique = true),
        }
)
public class Patient extends GenericModel<Integer> {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "patient_patient_id_seq")
    @SequenceGenerator(sequenceName = "patient_patient_id_seq", name = "patient_patient_id_seq", allocationSize = 1)
    @Column(name = "patient_id")
    private Integer id;
    @JoinColumn(name = "user_id", nullable = false)
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    private User user;
    @JoinColumn(name = "office_id", nullable = false)
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    private Office office;

    @Override
    public Integer getId() {
        return this.id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public User getUser() {
        return this.user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Office getOffice() {
        return this.office;
    }

    public void setOffice(Office office) {
        this.office = office;
    }

    @Override
    protected boolean isSameType(Object o) {
        return o instanceof Patient;
    }
}
